### DISCUSSION TITLE
Enable fine-grained opt-in for content.tabs.link

### DISCUSSION THREAD

**Discussion by chiphogg** [squidfunk/mkdocs-material#4792] [Ideas]
My goal is to be able to turn on content tab linking, but _only_ for specific tabs.  If I turn it on universally, I find I get linking in all kinds of undesired places.

This was tangentially discussed in #3982.  One suggestion was to provide a syntax similar to the fine-grained opt-in for annotations on code blocks: say, `=== "title" {.link}`.  I got the impression that this would be difficult or impossible to get working without breaking things.

An alternative might be to put the linked tabs in the config file.  This plays more nicely with their status as persistent _global_ links across the site (which I understand is the case).  So we could have something like:

```yaml
theme:
  features:
    - content.tabs.link:
      tabs_to_link:
        - clang14
        - gcc10
```

(I'm pretty shaky on my YAML syntax, but I hope that gives the general idea.)

With this feature, when a tab is clicked, instead of always linking, it would first be checked against the global allowlist defined above.

### Detailed example: 

To add some more colour: I'm building a doc site for a potentially open source C++ project.

One page is a "troubleshooting" page, which has examples of the most common compiler errors, and explanations of what they mean.  It's a long page, intended to let users `Ctrl-F` and enter some key pieces of an actual compiler error they encounter.  I would like to have a tab for each compiler: say, one for `clang14`, one for `gcc10`, etc.  These are _much_ better when linked: the user can select their preferred compiler, and all examples will switch simultaneously, hiding examples the user does not care about.

Another series of pages is the "tutorial" pages.  These have a bunch of `Problem` / `Solution` tabs.  I _really don't_ want these to be linked, because selecting the first solution will mean that the solution for the next one is automatically shown, rather than the problem.

**Comment by squidfunk** [comment]
Thanks for the feature suggestion, but I think it impacts only a handful of users. Let's let it sit here for a while and see if others might find it useful. However, IMHO, it might appear rather unpredictable to the user which tabs will link and which not. Since this is a request on a feature that is part of the community edition, forking and adjusting the relevant logic is always an option.

**Reply by chiphogg** [reply]
That's a good point about the unpredictability; I hadn't thought of that.

Maybe there's another way, where we could turn on linking at the _file_ level? I don't know if there's any precedent for any kind of per file setting in mkdocs.

**Reply by squidfunk** [reply]
There's currently no option to do so, as it would need to be integrated with [instant loading](https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#instant-loading), which is another problem. It might be possible, and I'm currently thinking about how to allow to switch off certain features for certain pages, but that's a huge problem given how many features and combinations of features Material for MkDocs has.
